proposals/flexible-vectors/FlexibleVectors.md

```diff
@@ -0,0 +1,327 @@
+Flexible vectors overview
+=========================
+
+The goal of this proposal is to provide flexible vector instructions for
+WebAssembly as a way to bridge the gap between existing SIMD instruction sets
+available on various platforms. `simd128` proposal already identified
+operations that would commonly work on platforms that are important to
+WebAssembly, this proposal is attempting to extend the same operations to work
+with variable vector lengths.
+
+## Types
+
+Proposal introduces the following vector types:
+
+- `vec.v8` : 8-bit lanes
+- `vec.v16`: 16-bit lanes
+- `vec.v32`: 32-bit lanes
+- `vec.v64`: 64-bit lanes
+
+### Lane division interpretation
+
+In semantic pseudocode `S` is the particular vector type, `S.LaneBits` is the
+size of the lane in bits, `S.Lanes` is the number of lanes, which is dynamic.
+
+|    S      | S.LaneBits |
+|-----------|-----------:|
+| `vec.v8`  |          8 |
+| `vec.v16` |         16 |
+| `vec.v32` |         32 |
+| `vec.v64` |         64 |
+
+### Restrictions
+
+Lane values are intended to be handled exactly like in `simd128` proposal, with
+the following differences applying to overall types:
+
+- Runtime sets maximum vector length for every type
+- Number of lanes is set separately for different lane sizes
+- Vectors with different lane size are not immediately interoperable
+
+## Immediate operands
+
+_TBD_ value range, depends on instruction encoding.
+
+- `ImmLaneIdxV8`: lane index for 8-bit lanes
+- `ImmLaneIdxV16`: lane index for 16-bit lanes
+- `ImmLaneIdxV32`: lane index for 32-bit lanes
+- `ImmLaneIdxV64`: lane index for 64-bit lanes
+
+## Operations
+
+Completely new operations introduced in this proposal are the operations that
+provide interface to vector length.
+
+### Vector length
+
+Querying length of supported vector:
+
+- `vec.v8.length -> i32`
+- `vec.v16.length -> i32`
+- `vec.v32.length -> i32`
+- `vec.v64.length -> i32`
+
+### Constructing vector values
+
+Create vector with identical lanes:
+
+- `vec.v8.splat(x:i32) -> vec.v8`
```

Comment:
arguments should not be all `i32`?

Reply:
`i32` is the only option, there is no `i8` type. The [same][s] is true for `simd128`. The value gets truncated I believe.

[s]: https://github.com/WebAssembly/simd/blob/master/proposals/simd/SIMD.md#create-vector-with-identical-lanes

Reply:
and `i64` ?

Reply:
True, thank you for pointing out.

Reply:
Fixed in the update